### ISSUE TITLE
AuthorizationRequest bug

### ISSUE THREAD

**Issue by GladFox** [Unity-Technologies/com.unity.mobile.notifications#37]
Added an event listener to the constructor, but forgot to release it into the event listener in Dispose

` public AuthorizationRequest(AuthorizationOption authorizationOption, bool registerForRemoteNotifications)
{
    ...
    iOSNotificationCenter.OnAuthorizationRequestCompleted += data =>
    ...
}
`

`public void Dispose()
 {
 }`

**Comment by GladFox** [comment]
If I run the program I execute the authorization request several times, then the callback will call several times. An authorization request has to be called up every time when it is necessary to enable notification, because the user could turn off notifications in the settings or vice versa enable them.

**Comment by GladFox** [comment]
I think it will be more correct

```
public AuthorizationRequest(AuthorizationOption authorizationOption, bool registerForRemoteNotifications)
        {
            iOSNotificationsWrapper.RegisterAuthorizationRequestCallback();
            iOSNotificationsWrapper.RequestAuthorization((int)authorizationOption, registerForRemoteNotifications);
            iOSNotificationCenter.OnAuthorizationRequestCompleted += OnAuthorizationRequestCompleted;
        }

		private void OnAuthorizationRequestCompleted(iOSAuthorizationRequestData data)
		{
            IsFinished = data.finished;
            Granted = data.granted;
            Error = data.error;
            DeviceToken = data.deviceToken;
		}

        public void Dispose()
        {
            iOSNotificationCenter.OnAuthorizationRequestCompleted -= OnAuthorizationRequestCompleted;
        }

```

**Comment by vincent-unity** [comment]
Thanks! And yes, we noticed this issue and will do a fix for it.

**Comment by GladFox** [comment]
Good news! Thank you!

**Comment by vincent-unity** [comment]
Btw, could you please report a bug with repro project anyway? Then our QA can help and verify this.

**Comment by vincent-unity** [comment]
Fixed in 1.2.0-preview.
